Code/src/main/java/nl/utwente/group10/ui/components/anchors/ConnectionAnchor.java

```diff
@@ -15,38 +21,134 @@
+    /** The visual representation of the ConnectionAnchor. */
+    @FXML private Shape visibleAnchor;
+    
+    /** The invisible part of the ConnectionAnchor (the touch zone). */
+    @FXML private Shape invisibleAnchor;
+    
+    /** Property storing the error state. */
+    private BooleanProperty errorState;
+    
+    /**
+     * Property storing the active state.
+     * When active, the ConnectionAnchor will react to user input.
+     */
+    private BooleanProperty activeState;
 
     /**
      * @param block
-     *            The block where this Anchor is connected to.
-     * @param pane
-     *            The pane this Anchor belongs to.
+     *            The block this ConnectionAnchor belongs to.
      */
-    public ConnectionAnchor(Block block, CustomUIPane pane) {
-        this.block = block;
-        this.pane = pane;
-
+    public ConnectionAnchor(Block block) {
         this.loadFXML("ConnectionAnchor");
+        
+        this.block = block;
+        this.errorState = new SimpleBooleanProperty(false);
+        this.activeState = new SimpleBooleanProperty(true);
         connections = new ArrayList<Connection>();
+        
+        activeState.addListener(a -> invalidateActive());
+        errorState.addListener(this::checkError);
+    }
+    
+    /**
+     * @return Whether or not this ConnectionAnchor is in active mode.
+     */
+    public boolean getActiveState() {
+        return activeState.get();
+    }
+    
+    /**
+     * @param active The new active state for this ConnectionAnchor.
+     */
+    public void setActiveState(boolean active) {
+        this.activeState.set(active);
+    }
+    
+    /**
+     * @return The property describing the active state of this ConnectionAnchor.
+     */
+    public BooleanProperty activeStateProperty() {
+        return activeState;
+    }
+    
+    /**
+     * @return Whether or not this ConnectionAnchor is in an error state.
+     */
+    public boolean getErrorState() {
+        return errorState.get();
+    }
+    
+    /**
+     * @param state The new error state for this ConnectionAnchor.
+     */
+    public void setErrorState(boolean state) {
+        errorState.set(state);
+    }
+    
+    /**
+     * @return The property describing the error state of this ConnectionAnchor.
+     */
+    public BooleanProperty errorStateProperty() {
+        return errorState;
+    }
+    
+    /**
+     * @return The Shape that is the visible part of the ConnectionAnchor.
+     */
+    public Shape getVisibleAnchor() {
+        return visibleAnchor;
+    }
+    
+    /**
+     * @return The Shape that is the invisible part (touch zone) of this ConnectionAnchor.
+     */
+    public Shape getInvisibleAnchor() {
+        return invisibleAnchor;
     }
 
     /**
+     * @return The Expr this ConnectionAnchor represents (coming from a Block).
+     */
+    public abstract Expr getExpr();
+    
+    /**
      * @return Input or output type of the block associated with this anchor.
```

Comment:
...or an empty Optional when any kind of error occurs. You could add that to the Javadoc.